test/e2e/telemeter_client_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023 The Cluster Monitoring Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/openshift/cluster-monitoring-operator/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// TestTelemeterClient verifies that the monitoring stack can get data on
+// telemeter client from /federate endpoint
+
+func TestTelemeterClient(t *testing.T) {
+	{
+		cm := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      clusterMonitorConfigMapName,
+				Namespace: f.Ns,
+				Labels: map[string]string{
+					framework.E2eTestLabelName: framework.E2eTestLabelValue,
+				},
+			},
+			Data: map[string]string{
+				"config.yaml": "{}",
+			},
+		}
+		f.MustCreateOrUpdateConfigMap(t, cm)
+		t.Cleanup(func() {
+			f.MustDeleteConfigMap(t, cm)
+		})
+
+		ctx := context.Background()
+		patch := []byte(`{
+	"spec": {
+		"overrides": [{
+			"group": "apps",
+			"kind": "Deployment",
+			"name": "cluster-monitoring-operator",
+			"namespace": "openshift-monitoring",
+			"unmanaged": false
+		}]
+	}
+}`)
+		_, err := f.OpenShiftConfigClient.ConfigV1().ClusterVersions().Patch(ctx, "version", types.MergePatchType, patch, metav1.PatchOptions{})
```

Comment:
You shouldn't have to put CMO into unmanaged state? IIUC the test only needs to check that telemeter client collects metrics from Prometheus.

Reply:
its unmanaged: false to diferenciate the other test that is unmanaged: true
But I can delete de line if its clearer

Reply:
its unmanaged: false to diferenciate the other test that is unmanaged: true
But I can delete de line if its clearer

Reply:
done

Reply:
yes please, I think that you only need to keep the code starting at L71. Also the test case could go in test/e2e/test_telemeter.go

Reply:
The idea is to decouple tests as this one is testing telemeter-client and the other one prometheus remoteWrite. In the future when telemeter client is not needed it will clearer delete this test.
In any case, if needed I can move it to test_telemeter.go